#57169 fix/streamResetBurst-and-streamResetRate-options-are-missing-in-the-documentation-of-http2.CreateSecureServer

Sujal-Raj opened the pull request:
<!--
Before submitting a pull request, please read:

- the CONTRIBUTING guide at https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md
- the commit message formatting guidelines at
  https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

If you believe this PR should be highlighted in the Node.js CHANGELOG
please add the `notable-change` label.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

Title: Documentation Update: Include streamResetBurst and streamResetRate Options in http2.CreateSecureServer

Description:

This PR addresses the issue where the streamResetBurst and streamResetRate options were missing in the documentation for http2.CreateSecureServer.

Changes:

Added documentation for the streamResetBurst and streamResetRate options in the http2.CreateSecureServer section.


If you need any adjustments or further additions, feel free to let me know!

lpinca commented on doc/api/http2.md:
```diff
@@ -3055,6 +3055,9 @@ changes:
     **Default:** `100`.
   * `settings` {HTTP/2 Settings Object} The initial settings to send to the
     remote peer upon connection.
+  * `streamResetBurst` {number} and `streamResetRate` {number} Sets the rate
```
I think a separate bullet point, one for each option, would be better. I understand that both should be set, but it can be specified in the description.

Sujal-Raj replied:
@lpinca  Thank you for the feedback! I appreciate the suggestion to use separate bullet points for each option. But if we look above in the http2.createServer it is written same as the above. So to maintain that flow of document. I think this goes good. However if the separate bullet point  would be better i can make the necessary changes .

pimterry reviewed (commented):
Hi @Sujal-Raj! It looks like the CI checks are failing here, with linting issues on the commit message & the markdown formatting.

To tidy that up so we can merge this, you'll need to update the commit message to match the required format, and fix the formatting. There's more details in the PR guide [here](https://github.com/nodejs/node/blob/main/doc/contributing/pull-requests.md) which is worth a read. The key parts are the docs [here](https://github.com/nodejs/node/blob/main/doc/contributing/pull-requests.md#commit-message-guidelines) regarding commit message formats, and then running `make lint` locally to check & fix the formatting on your machine.

Sujal-Raj commented:
@pimterry I've updated the commit message and fixed the markdown formatting, and all CI checks have passed.  Let me know if there's anything else needed! Thanks!

pimterry commented:
Ah, no, the commit message now passes but it looks like the markdown linting is still failing @Sujal-Raj. Can you take a look? You should be able to autofix it by running `make format-md` locally. Let me know if you have any questions at all.

Sujal-Raj commented:
@pimterry I’ve run make format-md locally and committed the changes. Let me know if there’s anything else I should check. Thanks!
